src/main/java/demo/CreateIndexWorkflowStep.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+package demo;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.opensearch.action.admin.indices.create.CreateIndexResponse;
+import org.opensearch.flowframework.workflow.WorkflowData;
+import org.opensearch.flowframework.workflow.WorkflowStep;
+
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.CompletableFuture;
+
+/**
+ * Sample to show other devs how to pass data around.  Will be deleted once other PRs are merged.
+ */
+public class CreateIndexWorkflowStep implements WorkflowStep {
+
+    private static final Logger logger = LogManager.getLogger(CreateIndexWorkflowStep.class);
+
+    private final String name;
+
+    /**
+     * Instantiate this class.
+     */
+    public CreateIndexWorkflowStep() {
+        this.name = "CREATE_INDEX";
+    }
+
+    @Override
+    public CompletableFuture<WorkflowData> execute(List<WorkflowData> data) {
+        CompletableFuture<WorkflowData> future = new CompletableFuture<>();
+        CompletableFuture.runAsync(() -> {
```

Comment:
Haven't seen `CompletableFuture.runAsync()` before. Just wanted to understand the rationale for this step and if its necessary to handle each `WorkflowStep` this way

Reply:
In the future I'll be adding an executor argument to this to define which thread pool it will run on (see #42). It's much the same as ExecutorService.submit().

The rationale here is that we can launch this separate runnable and it will run in a different thread, and eventually complete `future` when it does its work.  In the meantime the `execute()` method returns the (uncompleted) future so the next step can wait for all the predecessors to complete with an allOf().